spec/system/support_interface/editing_immigration_status_non_eu_spec.rb

```diff
@@ -0,0 +1,136 @@
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Skilled Worker visa')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Skilled Worker visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Dependent on partner’s or parent’s visa')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Dependent on partner’s or parent’s visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Family visa')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Family visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('British National (Overseas) visa')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('British National (Overseas) visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('UK Ancestry visa')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('UK Ancestry visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('High Potential Individual visa')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('High Potential Individual visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Youth Mobility Scheme')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Youth Mobility Scheme')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('India Young Professionals Scheme visa')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('India Young Professionals Scheme visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Ukraine Family Scheme or Ukraine Sponsorship Scheme visa')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Ukraine Family Scheme or Ukraine Sponsorship Scheme visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Afghan citizens resettlement scheme or Afghan Relocations and Assistance Policy')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Afghan citizens resettlement scheme or Afghan Relocations and Assistance Policy')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Refugee Status')
+    and_i_add_an_audit_comment
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Refugee Status')
+  end
+
+  def and_i_add_an_audit_comment
```

Comment:
Can you sort these methods in the order they appear in the test? I've started trying to do this when I can remember to.

Reply:
Interesting. My thoughts on this are: 

- It would be nice to read down the page and see exactly what the test is doing easily in one go.   
- What happens in specs where there are more than one scenario?
- This could also get messy and take time in large files?
- I'm not sure this provides much value for me personally, I always search for the method?


Either way I’ve done it on this occasion here: https://github.com/DFE-Digital/apply-for-teacher-training/pull/9012/commits/22512a69ee5c9d4d346eb342197faeb2321cee26